apps/api/src/server/migration/migrators/service-user-migrator.js

```diff
@@ -0,0 +1,118 @@
+import { MigratedEntityIdCeiling } from '../migrator.consts.js';
+import { buildUpsertForEntity } from './sql-tools.js';
+import { databaseConnector } from '#utils/database-connector.js';
+
+/**
+ * @typedef import('pins-data-model').Schemas.ServiceUser ServiceUser
+ */
+
+/**
+ * @param {ServiceUser[]} serviceUsers
+ */
+export const migrateServiceUsers = async (serviceUsers) => {
+	console.info(`Migrating ${serviceUsers.length} Service Users`);
+
+	for (const serviceUser of serviceUsers) {
+		const serviceUserEntity = mapModelToServiceUserEntity(serviceUser);
+
+		if (serviceUserEntity.id >= MigratedEntityIdCeiling) {
+			throw Error(`Unable to migrate entity id=${serviceUserEntity.id} - identity above threshold`);
+		}
+
+		const { statement: serviceUserStatement, parameters: serviceUserParameters } =
+			buildUpsertForEntity('ServiceUser', serviceUserEntity, 'id');
+
+		console.info(`Upserting ServiceUser with ID ${serviceUserEntity.id}`);
+
+		await databaseConnector.$transaction([
+			databaseConnector.$executeRawUnsafe(serviceUserStatement, ...serviceUserParameters)
+		]);
+
+		const serviceUserRecord = await databaseConnector.serviceUser.findUnique({
+			where: { id: serviceUserEntity.id }
+		});
+		const addressEntity = mapModelToAddressEntity(serviceUser);
```

Comment:
I _think_ we may be able to do this with Prisma (although it will essentially perform the same set of statements). Does this work?

```javascript
const updatedAddress = await databaseConnector.address.upsert({
  where: {
    ServiceUser: {
      some: {
        id: serviceUserId,
      },
    },
  },
  update: {
    ...newAddressData,
  },
  create: {
    ...newAddressData,
    ServiceUser: {
      connect: { id: serviceUserId },
    },
  },
});
```